src/Solr/Reindex/Handlers/SolrReindexBase.php

```diff
@@ -224,7 +238,19 @@ protected function getRecordsInGroup(SolrIndex $indexInstance, $class, $groups,
             $items = $items->filter('ClassName', $class);
         }
 
-        return $items;
+        $indexableService = IndexableService::singleton();
+
+        // ShowInSearch filter
+        // we cannot use $items->remove($item), as that deletes the record from the database
+        $ids = $items->map('ID', 'ID')->toArray();
+        foreach ($items as $item) {
+            if (!$indexableService->isIndexable($item)) {
+                unset($ids[$item->ID]);
+            }
+        }
+        // If no $ids, then filter against 0 to prevent exception: Cannot filter "SiteTree"."ID" against an empty set
+        $value = !empty($ids) ? array_keys($ids) : 0;
+        return DataList::create($class)->filter(['ID' => $value]);
```

Comment:
This feels like a bit of a foot gun: You're re-creating the $items list above as a separate DataList, rather than filtering out from the existing list. I'd reframe this to `$items->exclude('ID', $removedIds)`

Reply:
Have changed it to use exclude()